### PR TITLE
Elasticsearch template improvements

### PR DESCRIPTION
**Pull request by hglkrijger**
- add a new test plan for query performance monitoring
- add support for public ssh key authentication in linux
- add support for the cloud-azure plugin in linux

### CODE REVIEW

File: elasticsearch/azuredeploy.json [hunk 1]
```diff
@@ -247,6 +266,31 @@
...
     },
+    "cloudAzure" : {
+      "type": "string",
+      "defaultValue": "no",
+      "allowedValues": [
+          "yes",
+          "no"
+      ],
+      "metadata": {
+          "description": "Install the cloud-azure plugin"
```
Comment (singhkay) [inline elasticsearch/azuredeploy.json]: Can you add a documentation link of what cloud-azure is?

File: elasticsearch/azuredeploy.json [hunk 2]
```diff
@@ -391,6 +460,7 @@
         "sku": "14.04.2-LTS",
```
Comment (singhkay) [inline elasticsearch/azuredeploy.json]: I recommend updating to 14.04.4 LTS

File: elasticsearch/azuredeploy.json [hunk 3]
```diff
@@ -408,7 +478,7 @@
             "typeHandlerVersion": "1.2",
```
Comment (singhkay) [inline elasticsearch/azuredeploy.json]: Also update to 1.4

**Comment by hglkrijger** [comment]
Updated the image to 14.04-4, extension to 1.4 and added some more description on the plugin

**Comment by singhkay** [comment]
lgtm, merged

**Pull request merged**